### DISCUSSION TITLE
How can I change the base path in fumadoc-mdx@10?

### DISCUSSION THREAD

**Discussion by nktnet1** [fuma-nama/fumadocs#813] [Q&A] [answered]
After updating to the v10 release of fumadocs-mdx, I can no longer configure the base path:
- https://github.com/fuma-nama/fumadocs/releases/tag/fumadocs-mdx%4010.0.0

Here is an [example repository](https://github.com/nktnet-bug-report/fumadocs-test-app) - the only thing I changed after running
```
npm create fumadocs-app
```
is this commit [here](https://github.com/nktnet-bug-report/fumadocs-test-app/commit/b75d518df3011c343183d8ccb237145fed749c89), where all references of `/docs` was changed to `/24T3`.

Going to `https://localhost:3000/24T3` gives me 404. Did something else changed? I've also tried re-generating files with the postinstall script.

**Comment by fuma-nama** [comment]
Change source config file

**Reply by nktnet1** [reply]
Hey @fuma-nama, this is what my [source.config.ts](https://github.com/nktnet-bug-report/fumadocs-test-app/blob/main/source.config.ts) looks like (generated from the default):
```ts
import { defineDocs, defineConfig } from 'fumadocs-mdx/config';

export const { docs, meta } = defineDocs();

export default defineConfig();
```

Which option would I need to change to move `/docs` to `/24T3`?

It wasn't too clear to me looking at:
- https://fumadocs.vercel.app/docs/mdx/configuration
- https://fumadocs.vercel.app/docs/mdx/configuration#default-mdx-options

**Reply by fuma-nama** [reply, marked as answer]
https://fumadocs.vercel.app/docs/mdx/configuration#define-docs

Give docs option a dir property, you can specify the content directory there

**Reply by nktnet1** [reply]
Ah that worked - thank you! I had to do `./content` instead of `./content/docs`

 `source.config.ts`:
```ts
import { defineDocs, defineConfig } from 'fumadocs-mdx/config';

export const { docs, meta } = defineDocs({
  docs: {
    dir: './content',
  },
});

export default defineConfig();
```

Thank you!

**Reply by nktnet1** [reply]
Hi @fuma-nama,

Somewhat related question, will it be possible to render the docs at the root?

e.g.
```
/24T3
/25T1
/25T2
... etc
```

---

Ideally I'd want it to look similar to your fumadocs website, but rather than
```
https://fumadocs.vercel.app/docs/headless
https://fumadocs.vercel.app/docs/ui
https://fumadocs.vercel.app/docs/mdx
```
remove the `/docs` part, i.e.

```
https://fumadocs.vercel.app/headless
https://fumadocs.vercel.app/ui
https://fumadocs.vercel.app/mdx
```

However, I still want to be able to use other routes such as
```
https://fumadocs.vercel.app/login
https://fumadocs.vercel.app/profile
https://fumadocs.vercel.app/api/auth/logout
etc
```

I've tried messing with the different configurations and tweaking your site source code as well, but couldn't get it to work :(. Any suggestions would be deeply appreciated :pray:!

Thank you!!

**Reply by fuma-nama** [reply]
Just set base url to '/', and make /app/docs a route group: /app/(docs)

You don't need to change the other pages. Make sure to read next.ja docs to learn more about routing

**Reply by nktnet1** [reply]
Thank you :D!

I finally got it to work! Some small stuff I picked up on along the way:

---

## Issue  1

Typo in "you cna configure" (the word "can") in `./app/layout.config.tsx` that was generated from `npm create fumadocs-app`:

![image](https://github.com/user-attachments/assets/cdc4f4ce-7e96-4150-a8cb-08ec7f29e6ee)

---

## Issue 2

Docs for [root-toggle](https://fumadocs.vercel.app/docs/ui/components/root-toggle) shows

![image](https://github.com/user-attachments/assets/2f71dca9-9cca-4bb3-9ee5-2766f8719ae9)

but the "icons" property is actually "required"
![image](https://github.com/user-attachments/assets/db10d331-2d4e-4e12-ae76-541eb5495412)

---

**Issue 3**

Not sure if this is a bug or some misconfiguration on my end, but placing `.md` and `.mdx` inside of `./content/docs` work as usual with the source.config.ts:
```ts
import { defineDocs, defineConfig } from 'fumadocs-mdx/config';

export const { docs, meta } = defineDocs({
  docs: {
    dir: './content/docs'
  }
});

export default defineConfig();
```

However, when I renamed `./content/docs` to `./content/terms`, and changed the "dir" propertiy above to `'./content/terms`, everything still worked, but the `meta.json` files of every directory (and sub-directories) stopped working (i.e. not being used).

---

Do you have any idea what may have caused issue 3?

**Reply by fuma-nama** [reply]
Issue 3: you forgot to change the configurations for `meta`

**Reply by nktnet1** [reply]
oh - do you mind pointing me to the right docs for this? I'd like to rename it to /terms instead of /docs if possible!

I've looked through:
- https://fumadocs.vercel.app/docs/headless/page-conventions#meta
- https://fumadocs.vercel.app/docs/mdx/configuration#type

but wasn't sure where to change it to make fumadocs pick up on all meta.json files

**Reply by nktnet1** [reply]
I figured it out :D

```ts
import { defineConfig, defineDocs } from 'fumadocs-mdx/config';

const TERMS_DIRECTORY = './content/terms';

export const { docs, meta } = defineDocs({
  meta: {
    dir: TERMS_DIRECTORY,
  },
  docs: {
    dir: TERMS_DIRECTORY,
  },
});

export default defineConfig();
```

Thanks @fuma-nama!

**Reply by shubhkpatel** [reply]
By default, all mdx content starts with "/docs". Is there any way through which all docs url start with "/" ?

When I change baseUrl in source.ts, I am getting 404 for all mdx contents

**Reply by fuma-nama** [reply]
Of course, Because you didn't change the routing structure from app folder
